allennlp/data/dataset_readers/brown_corpus.py

```diff
@@ -0,0 +1,79 @@
+import tqdm
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.data.dataset import Dataset
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers.token_indexer import TokenIndexer
+from allennlp.data.fields import TextField, TagField
+from allennlp.data.token_indexers import SingleIdTokenIndexer
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("brown_corpus")
+class BrownCorpusDatasetReader(DatasetReader):
+    """
+    Reads instances from the Brown Corpus where each line is in the following format:
+
+    WORD/TAG WORD/TAG WORD/TAG ... WORD/TAG
+
+    and converts it into a ``Dataset`` suitable for sequence tagging.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.  See :class:`TokenIndexer`.
+        Note that the `output` tags will always correspond to single token IDs based on how they
+        are pre-tokenised in the data file.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    @overrides
+    def read(self, file_path):
+        with open(file_path, "r") as data_file:
+
+            instances = []
+            logger.info("Reading instances from lines in file at: %s", file_path)
+            for line in tqdm.tqdm(data_file):
+                line = line.strip()
+                if not line:  # skip blank lines
+                    continue
+
+                tokens_and_tags = [pair.split("/") for pair in line.split()]
+                tokens = [x[0] for x in tokens_and_tags]
+                tags = [x[1] for x in tokens_and_tags]
+
+                sequence = TextField(tokens, self._token_indexers)
+                sequence_tags = TagField(tags, sequence)
+                instances.append(Instance({'tokens': sequence,
+                                           'tags': sequence_tags}))
+        if not instances:
+            raise ConfigurationError("No instances were read from the given filepath {}. "
+                                     "Is the path correct?".format(file_path))
+        return Dataset(instances)
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'BrownCorpusDatasetReader':
+        """
+        Parameters
+        ----------
+        token_indexers: ``Dict[Params]``, optional
+        """
+        token_indexers = {}
+        token_indexer_params = params.pop('token_indexers', Params({}))
+        for name, indexer_params in token_indexer_params.items():
+            token_indexers[name] = TokenIndexer.from_params(indexer_params)
+        # The default parameters are contained within the class,
```

Comment:
Looks like all of our readers have this logic duplicated - can we put it somewhere central and re-use it?  Either on `TokenIndexer` as a method like `indexer_dict_from_params`, or something, or in `DatasetReader` as a helper function to process these arguments.  Also happy for this to be raised as a new issue if you don't want to deal with it in this PR.